apps/mp/services.py

```diff
@@ -571,7 +567,9 @@ def parse_dependency(model, dependency):
                 is_reverse = field.__class__.__name__.startswith('Reverse')
                 field = field.field
                 model = field.rel.related_model if is_reverse else field.rel.to
-                related_name = field.rel.related_name
+                related_name = (
+                    field.name if is_reverse else field.rel.related_name
+                )
```

Comment:
# Reviewer Note
Main Change!
The `related_name` should be current attribute name in reverse relations. `field.rel.related_name` points to next Model in path.